### PR TITLE
Fix kie demo

### PR DESCRIPTION
**Pull request by innerlee**
Fix https://github.com/open-mmlab/mmocr/issues/599

Verify:
```bash
./mmocr/utils/ocr.py --config-dir configs --det PS_CTW --recog SAR --kie SDMGR --print-result --output . --kie-config configs/kie/sdmgr/sdmgr_novisual_60e_wildreceipt.py --kie-ckpt https://download.openmmlab.com/mmocr/kie/sdmgr/sdmgr_novisual_60e_wildreceipt_20210517-a44850da.pth  demo/demo_kie.jpeg
```

### CODE REVIEW

File: mmocr/utils/ocr.py [hunk 1]
```diff
@@ -508,10 +509,16 @@ def visualize_kie_output(self,
         img_tensor = data['img'].data
         img_meta = data['img_metas'].data
         gt_bboxes = data['gt_bboxes'].data.numpy().tolist()
-        img = tensor2imgs(img_tensor.unsqueeze(0),
-                          **img_meta['img_norm_cfg'])[0]
-        h, w, _ = img_meta['img_shape']
-        img_show = img[:h, :w, :]
+        if img_tensor.dtype == torch.uint8:
```
Comment (gaotongxiao) [inline mmocr/utils/ocr.py:512]:
```suggestion
        if img_tensor.dtype == torch.uint8:
            # The img tensor is the raw input not being normalized
            # (For SDMGR non-visual)
```

File: mmocr/utils/ocr.py [hunk 2]
```diff
@@ -508,10 +509,16 @@ def visualize_kie_output(self,
...
+            img = img_tensor.cpu().numpy().transpose(1, 2, 0)
+        else:
+            img = tensor2imgs(
+                img_tensor.unsqueeze(0), **img_meta.get('img_norm_cfg', {}))[0]
+        if 'img_shape' in img_meta:
+            h, w, _ = img_meta['img_shape']
+            img_show = img[:h, :w, :]
+        else:
+            img_show = img
```
Comment (gaotongxiao) [inline mmocr/utils/ocr.py:521]:
```suggestion
        h, w, _ = img_meta.get('img_shape', img.shape)
        img_show = img[:h, :w, :]
```

File: mmocr/utils/ocr.py [hunk 3]
```diff
@@ -593,6 +600,11 @@ def det_recog_kie_inference(self, det_model, recog_model, kie_model=None):
                         min_x, min_y, max_x, min_y, max_x, max_y, min_x, max_y
                     ]
                 ann_info = kie_dataset._parse_anno_info(annotations)
+                if 'ori_boxes' not in ann_info:
+                    ann_info['ori_bboxes'] = ann_info['bboxes']
+                if 'gt_bboxes' not in ann_info:
+                    ann_info['gt_bboxes'] = ann_info['bboxes']
+                    # img_meta
```
Comment (gaotongxiao) [inline mmocr/utils/ocr.py:607]:
```suggestion
                ann_info['ori_bboxes'] = ann_info.get('ori_bboxes',
                                                      ann_info['bboxes'])
                ann_info['gt_bboxes'] = ann_info.get('gt_bboxes',
                                                     ann_info['bboxes'])
```

File: mmocr/utils/ocr.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (innerlee) [reply mmocr/utils/ocr.py:1]: Windows is an application platform, and seldom be an dev platform. Plus that win has wsl2.

[the remark this answers is not in the document]
Comment (gaotongxiao) [reply mmocr/utils/ocr.py:1]: Right, but we do want to extend the support to Windows. MMCV has been putting some effort into pre-compiling packages for Windows. Most of OpenMMLab repos will have Windows CI soon.

**Review by gaotongxiao — APPROVED** [review]

**Pull request merged**